### PR TITLE
read-write texture feature

### PR DESCRIPTION
**Pull request by eugenebokhan**

### CODE REVIEW

File: Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift [hunk 1]
```diff
@@ -15,6 +16,24 @@ public extension MTLDevice {
             #elseif os(macOS)
             return self.supportsFeatureSet(.macOS_GPUFamily1_v3)
             #endif
+            
+        case let .readWriteTexture(pixelFormat):
```
Comment (s1ddok) [inline Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:20]: I don't think that read_write texture is only pixel format dependent. It is surely also depends on GPU family

Comment (eugenebokhan) [reply Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:20]: yup, we get readWriteTextureSupport from current device and check if it supports this feature

Comment (s1ddok) [reply Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:20]: oh sorry I got it

File: Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift [hunk 2]
```diff
@@ -15,6 +16,24 @@ public extension MTLDevice {
...
+                .rgba16Uint, .rgba16Sint, .rgba8Unorm, .rgba8Uint,
+                .rgba8Sint, .r16Float, .r16Uint, .r16Sint,
+                .r8Unorm, .r8Uint, .r8Sint
+            ])
+            
+            switch self.readWriteTextureSupport {
+            case .tier1: return tierOneSupportedPixelFormats.contains(pixelFormat)
+            case .tier2: return tierTwoSupportedPixelFormats.contains(pixelFormat)
+            case .tierNone: return false
+            @unknown default: return false
```
Comment (s1ddok) [inline Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:35]: is it safe to add this? especially with false clause? I think all new cases should correspond to more modern designs

Comment (eugenebokhan) [reply Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:35]: im not sure about the desktop GPUs

File: Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift [hunk 3]
```diff
@@ -2,6 +2,7 @@ import Metal
 
 public enum Feature {
     case nonUniformThreadgroups
+    case readWriteTexture(MTLPixelFormat)
```
Comment (s1ddok) [inline Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:5]: should we call it texture*s*?

Comment (eugenebokhan) [reply Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:5]: `readWriteTextures` looks better

Comment (s1ddok) [reply Sources/Alloy/Core/Extensions/Metal/MTLDevice+Features.swift:5]: yes it exactly what I mean

**Pull request merged**